x-pack/plugins/apm/server/lib/service_map/get_service_map_from_trace_ids.ts

```diff
@@ -29,21 +25,19 @@ export function getConnections(
   if (serviceName || environment) {
     paths = paths.filter((path) => {
       return path.some((node) => {
-        let matches = true;
-        if (serviceName) {
-          matches =
-            matches &&
-            SERVICE_NAME in node &&
-            (node as ServiceConnectionNode)[SERVICE_NAME] === serviceName;
+        if (serviceName && node[SERVICE_NAME] !== serviceName) {
+          return false;
         }
-        if (environment && environment !== ENVIRONMENT_NOT_DEFINED.value) {
-          matches =
-            matches &&
-            SERVICE_ENVIRONMENT in node &&
-            (node as ServiceConnectionNode)[SERVICE_ENVIRONMENT] ===
-              environment;
+
+        if (environment) {
+          if (environment === ENVIRONMENT_NOT_DEFINED.value) {
+            return node[SERVICE_ENVIRONMENT] === null;
```

Comment:
I saw that when `ENVIRONMENT_NOT_DEFINED` is selected the node has the service.environment = null. thoughts?

Reply:
I don't we should _only_ check for `node[SERVICE_ENVIRONMENT] === null`. I think we should also check for `SERVICE_ENVIRONMENT in node`. I don't think the script we use guarantees the former (at least not as far as I can tell).

Reply:
But if the `SERVICE_ENVIRONMENT` is not available in node, this `node[SERVICE_ENVIRONMENT]` will return undefined, which in the end will return false too.

Reply:
To clarify, both `node[SERVICE_ENVIRONMENT] === null` and `!(SERVICE_ENVIRONMENT in node)` should result in a match when `environment === ENVIRONMENT_NOT_DEFINED.value`. You could also simplify this check by return `!node[SERVICE_ENVIRONMENT]`.

Reply:
perfect, thanks for the explanation!!!!